src/lib/blkalloc/blk_allocator.h

```diff
@@ -65,37 +63,28 @@ struct BlkAllocConfig {
     const uint32_t m_align_size;
     const blk_num_t m_capacity;
     const blk_num_t m_blks_per_portion;
+    const bool m_persistent{false};
     const std::string m_unique_name;
-    bool m_auto_recovery{false};
-    bool m_realtime_bm_on{false}; // only specifically turn off in BlkAlloc Test;
 
 public:
-    BlkAllocConfig(uint32_t blk_size, uint32_t align_size, uint64_t size, const std::string& name = "",
-                   bool realtime_bm_on = true) :
+    BlkAllocConfig(uint32_t blk_size, uint32_t align_size, uint64_t size, bool persistent,
+                   const std::string& name = "") :
             m_blk_size{blk_size},
             m_align_size{align_size},
             m_capacity{static_cast< blk_num_t >(size / blk_size)},
             m_blks_per_portion{std::min(HS_DYNAMIC_CONFIG(blkallocator.num_blks_per_portion), m_capacity)},
-            m_unique_name{name} {
-#ifdef _PRERELEASE
-        // for pre-release build, take it from input which is defaulted to true;
-        m_realtime_bm_on = realtime_bm_on;
-#else
-        // for release build, take it from dynamic config which is defaulted to false
-        m_realtime_bm_on = HS_DYNAMIC_CONFIG(blkallocator.realtime_bitmap_on);
-#endif
-    }
+            m_persistent{persistent},
+            m_unique_name{name} {}
 
     BlkAllocConfig(BlkAllocConfig const&) = default;
     BlkAllocConfig(BlkAllocConfig&&) noexcept = delete;
     BlkAllocConfig& operator=(BlkAllocConfig const&) = default;
     BlkAllocConfig& operator=(BlkAllocConfig&&) noexcept = delete;
     virtual ~BlkAllocConfig() = default;
-    void set_auto_recovery(bool is_auto_recovery) { m_auto_recovery = is_auto_recovery; }
 
     virtual std::string to_string() const {
-        return fmt::format("BlkSize={} TotalBlks={} BlksPerPortion={} auto_recovery={}", in_bytes(m_blk_size),
-                           in_bytes(m_capacity), m_blks_per_portion, m_auto_recovery);
+        return fmt::format("BlkSize={} TotalBlks={} BlksPerPortion={} persistent={}", in_bytes(m_blk_size),
+                           in_bytes(m_capacity), m_blks_per_portion, m_persistent);
```

Comment:
From allocator perspective the term `auto_recovery` doesn't mean much, but it cares whether the bitmap is persistent or not.